src/parquet/file_writer.h

```diff
@@ -79,6 +83,44 @@ class PARQUET_EXPORT RowGroupWriter {
   std::unique_ptr<Contents> contents_;
 };
 
+// RowGroupWriter implementation that allows bounding a RowGroup size to a certain value
+// Columns can be written in an order
+// All the values are compressed and stored in memory
+// Values are written to the file on Close
+
+class PARQUET_EXPORT RowGroupWriter2 {
+ public:
+  // Forward declare a virtual class 'Contents' to aid dependency injection and more
+  // easily create test fixtures
+  // An implementation of the Contents class is defined in the .cc file
+  struct Contents {
+    virtual ~Contents() = default;
+    virtual int num_columns() const = 0;
+    virtual int64_t num_rows() const = 0;
+    virtual int64_t current_compressed_bytes() const = 0;
+    virtual ColumnWriter* get_column(int i) const = 0;
```

Comment:
If the ColumnWriter can indirectly touch the RowGroupWriter's internal state it'd be best to return a `const ColumnWriter*` from the const accessor.

Reply:
will do!